src/test/java/de/tum/in/www1/artemis/AchievementIntegrationTest.java

```diff
@@ -0,0 +1,137 @@
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.AchievementService;
+import de.tum.in.www1.artemis.util.DatabaseUtilService;
+import de.tum.in.www1.artemis.util.RequestUtilService;
+
+public class AchievementIntegrationTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    @Autowired
+    DatabaseUtilService database;
+
+    @Autowired
+    RequestUtilService request;
+
+    @Autowired
+    AchievementService achievementService;
+
+    @Autowired
+    AchievementRepository achievementRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    CourseRepository courseRepository;
+
+    private User user;
+
+    private Course course;
+
+    private Achievement achievement;
+
+    @BeforeEach
+    public void initTestCase() {
+        user = database.addUsers(1, 0, 0).get(0);
+        course = database.addEmptyCourse();
+
+        achievement = new Achievement();
+        achievement.setId(1L);
+        achievement.setTitle("Test Achievement");
+        achievement.setDescription("Create correct relations");
+        achievement.setIcon("");
+        achievement.setRank(1);
+        achievement.setCourse(course);
+
+        achievement = achievementRepository.save(achievement);
+    }
+
+    @AfterEach
+    public void resetDatabase() {
+        database.resetDatabase();
+    }
+
+    @Test
+    public void testManyToManyRelationToUser() {
+        assertThat(user.getAchievements().size()).isEqualTo(0).as("Number of achievements for user should be 0");
+
+        user.addAchievement(achievement);
+        assertThat(user.getAchievements().size()).isEqualTo(1).as("Number of achievements for user should be 1");
+        assertThat(user.getAchievements().contains(achievement)).isTrue().as("User has correct achievement");
+        assertThat(achievement.getUsers().size()).isEqualTo(1).as("Number of users for achievement should be 1");
+        assertThat(achievement.getUsers().contains(user)).isTrue().as("Achievement has correct user");
+
+        user.removeAchievement(achievement);
+        assertThat(user.getAchievements().size()).isEqualTo(0).as("Number of achievements for user should be 0");
+        assertThat(user.getAchievements().contains(achievement)).isFalse().as("User does not have removed achievement");
+        assertThat(achievement.getUsers().size()).isEqualTo(0).as("Number of users for achievement should be 0");
+        assertThat(achievement.getUsers().contains(user)).isFalse().as("Achievement does not have incorrect user");
+    }
+
+    @Test
+    @WithMockUser(value = "student1", roles = "USER")
+    public void testManyToManyRelationToUserRepository() throws Exception {
+        user.addAchievement(achievement);
+        user = userRepository.save(user);
+
+        var achievements = request.get("/api/achievements", HttpStatus.OK, Set.class);
```

Comment:
You could use here the create REST Call instead and for the above lines

Reply:
There is no create REST call for achievements, only update.